aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -284,9 +296,12 @@ impl<T: Transaction, O: TransactionOutput<Txn = T>, E: Debug + Send + Clone + Ca
             .and_then(|txn_output| match txn_output.output_status() {
                 ExecutionStatus::Success(_) => None,
                 ExecutionStatus::SkipRest(_) => None,
-                ExecutionStatus::Abort(Error::UserError(err)) => Some(err.categorize()),
-                ExecutionStatus::Abort(Error::ModulePathReadWrite) => {
-                    Some(ErrorCategory::ValidError)
+                ExecutionStatus::Abort(_) => Some(ErrorCategory::ValidError),
```

Comment:
Do we use output_category anywhere else other than the one place in executor to defensive assert SpeculativeExecutionError must have already been a validation failure? If so, maybe we can simplify the implementation for now?